nixos/modules/installer/tools/nixos-option/nixos-option.cc

```diff
@@ -0,0 +1,599 @@
+    }
+}
+
+Value parseAndEval(EvalState * state, std::string const & expression, std::string const & path)
+{
+    Value v{};
+    state->eval(state->parseExprFromString(expression, absPath(path)), v);
+    return v;
+}
+
+void printValue(Context * ctx, Out & out, std::variant<Value, std::exception_ptr> maybe_value,
+                std::string const & path);
+
+void printList(Context * ctx, Out & out, Value & v)
+{
+    Out list_out(out, "[", "]", v.listSize());
+    for (unsigned int n = 0; n < v.listSize(); ++n) {
+        printValue(ctx, list_out, *v.listElems()[n], "");
+        list_out << Out::sep;
+    }
+}
+
+void printAttrs(Context * ctx, Out & out, Value & v, std::string const & path)
+{
+    Out attrs_out(out, "{", "}", v.attrs->size());
+    for (const auto & a : v.attrs->lexicographicOrder()) {
+        std::string name = a->name;
+        attrs_out << name << " = ";
+        printValue(ctx, attrs_out, *a->value, appendPath(path, name));
+        attrs_out << ";" << Out::sep;
+    }
+}
+
+void multiLineStringEscape(Out & out, std::string const & s)
+{
+    int i;
+    for (i = 1; i < s.size(); i++) {
+        if (s[i - 1] == '$' && s[i] == '{') {
+            out << "''${";
+            i++;
+        } else if (s[i - 1] == '\'' && s[i] == '\'') {
+            out << "'''";
+            i++;
+        } else {
+            out << s[i - 1];
+        }
+    }
+    if (i == s.size()) {
+        out << s[i - 1];
+    }
+}
+
+void printMultiLineString(Out & out, Value const & v)
+{
+    std::string s = v.string.s;
+    Out str_out(out, "''", "''", Out::MULTI_LINE);
+    std::string::size_type begin = 0;
+    while (begin < s.size()) {
+        std::string::size_type end = s.find('\n', begin);
+        if (end == std::string::npos) {
+            multiLineStringEscape(str_out, s.substr(begin, s.size() - begin));
+            break;
+        }
+        multiLineStringEscape(str_out, s.substr(begin, end - begin));
+        str_out << Out::sep;
+        begin = end + 1;
+    }
+}
+
+void printValue(Context * ctx, Out & out, std::variant<Value, std::exception_ptr> maybe_value, std::string const & path)
+{
+    try {
+        if (std::holds_alternative<std::exception_ptr>(maybe_value)) {
+            std::rethrow_exception(std::get<std::exception_ptr>(maybe_value));
+        }
+        Value v = evaluateValue(ctx, &std::get<Value>(maybe_value));
+        if (ctx->state->isDerivation(v)) {
+            describeDerivation(ctx, out, v);
+        } else if (v.isList()) {
+            printList(ctx, out, v);
+        } else if (v.type == tAttrs) {
+            printAttrs(ctx, out, v, path);
+        } else if (v.type == tString && std::string(v.string.s).find('\n') != std::string::npos) {
+            printMultiLineString(out, v);
+        } else {
+            ctx->state->forceValueDeep(v);
+            out << v;
+        }
+    } catch (ThrownError & e) {
+        if (e.msg() == "The option `" + path + "' is used but not defined.") {
```

Comment:
Can you add a comment in lib/modules.nix that if this message is changed, then `nixos-option` needs to be updated as well.

Reply:
Done